plasma_framework/contracts/src/framework/registries/ExitGameRegistry.sol

```diff
@@ -1,17 +1,21 @@
 pragma solidity ^0.5.0;
 
+import "../Protocol.sol";
 import "../utils/Operated.sol";
 import "../utils/Quarantine.sol";
 
 contract ExitGameRegistry is Operated {
+    using Quarantine for Quarantine.Data;
+
     mapping(uint256 => address) private _exitGames;
     mapping(address => uint256) private _exitGameToTxType;
-    using Quarantine for Quarantine.Data;
+    mapping(uint256 => uint8) private _protocols;
```

Comment:
So you're proposing that we add an interface of `protocol()` in exit game.
And to get the protocol of a tx type, we do sth like `framework.exitGames(txType).protocol()` instead?

I wasn't aware of this option. but would like to know is this what you're proposing?

Reply:
yes, that's what I meant. I thought it was better to ask the exit game what protocol it supports, rather than storing it elsewhere. But maybe the exit game doesn't need to know what protocol it supports and it's better to keep it separate?
TBH I'm not totally sure which is best, so I'll leave it up to you

Reply:
> But maybe the exit game doesn't need to know what protocol it supports

well...it definitely knows, so it is part of the exit game knowledge already, just currently it is done implicitly.

hmm, I don't have a strong opinion on which is better actually. At one side I see it simplifies the `registerExitGame()` function. On the other side, getting info from exit game means we would have external call twice, first to framework to get exitGame and the from exitGame to get the protocol.

I will probably try to move toward your proposal for simplification first.

Reply:
I end up decide to be lazy ><. Current code structure of having two routers makes it not so fun on adding interfaces, as both would need to implement it(?) or....mock it in tests?

I will skip this for now then as it seems like we both are not strong on either side. Please let me know if you think we should still do that @kevsul .

Reply:
No, I think it's fine to leave it as it is